lib/MemoryAnalysisModules/FindSource.cpp

```diff
@@ -105,8 +105,13 @@ const Instruction *findNoAliasSource(const Value *v,
   if (!f)
     return NULL;
 
+  // FIXME: correctly ported from llvm13
+  // where hasAttribute(unsigned Index, Attribute::AttrKind Kind)
+  // However, why do we need index in the first place?
   if (!f->getAttributes().hasAttributeAtIndex(0, Attribute::NoAlias) &&
-      !isNoAliasCall(v))
+      // isNoAliasFn (deprecated) checked for noAlias attribute & allocation function
+      // isNoAliasCall checks only for noAlias attribute
+      !(isNoAliasCall(v) || isAllocationFn(v, &tli)))
```

Comment:
Are the two cases (get attributes vs is no alias call) the same thing?

Reply:
In terms of checking attributes, yes. isNoAliasFn has an additional allocation fcn check, which has been accordingly added.